### ISSUE TITLE
About the depth completion model

### ISSUE THREAD

**Issue by manurare** [paulengstler/invisible-stitch#15]
Hey,

Thanks for sharing this amazing work

From my understanding the depth model has been trained with warped versions of a canonical depth map and it has learned how to fill the gaps caused by the warping. The gaps usually mean unseen content "behind foreground" which are further away than the foreground. However, in the paper is mostly used for extrapolation (or outpainting? ). In the extrapolated areas depth could be either nearer or further than the existing part.

If we only look to fill gaps of depth due to discolussions: Is there a way to force depth should be further than what is in front?

**Comment by paulengstler** [comment]
The _gaps_, or masking patterns, are completely random during the training. That means parts of the background or foreground, however you define this, might be missing (see `\tilde{D}` in Fig. 2 of the paper). Crucially, with our setup, the shapes, dimensions, and extent should vaguely match what you are trying to inpaint.

At the core of our fine-tuning procedure is the idea to "recover" ZoeDepth with added benefits. In the implementation (https://github.com/paulengstler/invisible-stitch/blob/main/train.py#L98), you'll see that we only swap out the patch embedding layer. For these kinds of layers specifically, we want them to be as _general_ as they can be, meaning to work with any (reasonable) input we feed into them. Therefore, we opted for random (non-semantic) masking patterns.
